urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/utils/ConfigUtils.java

```diff
@@ -0,0 +1,101 @@
+package com.sinopec.utils;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Created with IntelliJ IDEA. User: Administrator Date: 13-10-23 Time: 下午4:23
+ * To change this template use File | Settings | File Templates.
+ */
+public class ConfigUtils {
+	private Properties propertie;
+	private FileInputStream inputFile;
+
+	public ConfigUtils() {
+	}
+
+	public ConfigUtils(String filePath) {
+		propertie = new Properties();
+		try {
+			inputFile = new FileInputStream(filePath);
+			propertie.load(inputFile);
+
+		} catch (FileNotFoundException e) {
+			LogUtils.getTrace(e);
+		} catch (IOException e) {
+			LogUtils.getTrace(e);
+		} finally {
+			try {
+				if (inputFile != null)
+					inputFile.close();
+			} catch (IOException e) {
+				LogUtils.getTrace(e); // To change body of catch statement use
+										// File | Settings | File Templates.
+			}
+		}
+	}
+
+	public int getInt(String key) {
+		return Integer.parseInt(getValue(key));
+	}
+
+	public String getString(String key) {
+		return getValue(key);
+	}
+
+	public Double getDouble(String key) {
+		return Double.parseDouble(getValue(key));
+	}
+
+	public int getInt(String fileName, String key) {
+		return Integer.parseInt(getValue(fileName, key));
+	}
+
+	public String getString(String fileName, String key) {
+		return getValue(fileName, key);
+	}
+
+	public Double getDouble(String fileName, String key) {
+		return Double.parseDouble(getValue(fileName, key));
+	}
+
+	private String getValue(String key) {
+		if (propertie.containsKey(key)) {
+			String value = propertie.getProperty(key);
+			return value;
+		} else
+			return null;
+	}
+
+	private String getValue(String fileName, String key) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.